### PR TITLE
feat(3d-tiles): support s2 bounding volume

### PR DESCRIPTION
**Pull request by mspivak-actionengine**

### CODE REVIEW

File: modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts
Comment (ibgreen) [inline modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts:1]: The filename is unreadable. `convert-s2-to-obb.ts` ?

File: modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts
Comment (ibgreen) [inline modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts:7]: Add at least one line JSDoc. Ideally one line JSDoc on every field in the type as well.

File: modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts
Comment (ibgreen) [inline modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts:13]: - What is `Cart`? Please type out. `LngLat`? `Cartographic`?
- Add JSDoc line
- Add typescript types

File: modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts
Comment (ibgreen) [inline modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts:24]: Type out...
```suggestion
export function convertS2BoundingVolumeToOBB(
```

File: modules/3d-tiles/src/lib/utils/s2/s2-geometry.ts
Comment (ibgreen) [inline modules/3d-tiles/src/lib/utils/s2/s2-geometry.ts:1]: Is this copied from deck.gl or is it an independent effort? Seeing this in multiple places, starting to feel like it is time to make an `@math.gl/s2` module. @Pessimistress what do you think about that?

**Review by ibgreen — COMMENTED** [review]
Nice work! 

- Please make a thorough pass on typing, comments, variable naming before landing, see inline comments.
- There are no tests, at least a minimal test case for converting an S2 cell to an OBB would make a big difference.

Remember that we write this code not primarily for ourselves, but to make it easy for the maintainers who come after us who have to try to understand we wrote. Anything we can do to improve readability, clarity, typing, comments, test cases will help the next reader.

I do think moving the s2 utils and the new s2toOBB function to math.gl could make a lot of sense. That would keep complexity down here.

File: modules/3d-tiles/src/lib/parsers/helpers/parse-3d-implicit-tiles.ts
Comment (belom88) [inline modules/3d-tiles/src/lib/parsers/helpers/parse-3d-implicit-tiles.ts:26]: The parse - stringify is not optimal approach. I suggest make a copy function. S2BoundingVolume object is not so long to do that.

**Review by ibgreen — APPROVED** [review]

**Review by belom88 — APPROVED** [review]

**Pull request merged**
